### PR TITLE
feat: account details ui

### PR DESCRIPTION
**Pull request by sshanzel**
# WIP ⚠️ 

## Changes
We have decided to revamp the account details UI to segregate the concerns relating to profile, security, and other kinds of settings.

Newly introduced components are mainly for reused elements in this context.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Comment by sshanzel** [comment]
There is no integration in this PR yet, but the core structure has been built already. So feel free to do an initial review if you will have some spare time.

cc @rebelchris @kirillkurko

**Review by kirillkurko — COMMENTED** [review]
Looks good

**Review by rebelchris — CHANGES_REQUESTED** [review]
Well on the way 🙏

**Comment by sshanzel** [comment]
@rebelchris looks like there were no major conflicts. Should be ready to be merged in our intermediate branch. Let me know once everything's clear with you! 🚢

File: packages/shared/src/components/auth/common.tsx [hunk 1]
```diff
@@ -46,6 +57,10 @@ export const providerMap: Record<string, Provider> = {
   },
 };
 
+export const getProviderMapClone = (
```
Comment (rebelchris) [inline packages/shared/src/components/auth/common.tsx:60]: Why do we need the clone?

File: packages/webapp/components/layouts/AccountLayout/common.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+interface AccountPageContainerProps {
+  title: string;
+  actions?: ReactNode;
+  children?: ReactNode;
+  className?: ClassName;
+  onBack?: () => void;
+}
+
+export const AccountPageContainer = ({
```
Comment (rebelchris) [inline packages/webapp/components/layouts/AccountLayout/common.tsx:89]: This should probably be it's own component.

File: packages/webapp/pages/account/security.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        className="mt-6 w-fit btn-secondary"
+        onClick={() => setActiveDisplay(Display.ChangeEmail)}
+      >
+        Change email
+      </Button>
+    );
+  };
+
+  return (
+    <TabContainer showHeader={false} controlledActive={activeDisplay}>
```
Comment (rebelchris) [inline packages/webapp/pages/account/security.tsx:91]: We can probably migrate these to separate individual components

**Review by rebelchris — APPROVED** [review]
Awesome work,
Nothing breaking, but some suggestions we can log as individual tasks for the intermediate branch.

File: packages/webapp/components/layouts/AccountLayout/common.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/common.tsx:89]: Totally agree. Let me migrate this one.

File: packages/shared/src/components/auth/common.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/common.tsx:60]: This is just so the original object won't get modified, we had areas where some properties had to be modified.

File: packages/webapp/components/layouts/AccountLayout/common.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/common.tsx:89]: Totally agree, this has become large enough.

File: packages/webapp/pages/account/security.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/webapp/pages/account/security.tsx:91]: 100% agree with that 🚢

Comment (rebelchris) [reply packages/webapp/pages/account/security.tsx:91]: Added a task for it.

File: packages/webapp/components/layouts/AccountLayout/common.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/webapp/components/layouts/AccountLayout/common.tsx:89]: Added a task for it.

**Pull request merged**